### PR TITLE
Fix a linked-list bug with the exclusivity runtime when accesses are not strictly nested

### PR DESCRIPTION
**Pull request by rjmccall**

### CODE REVIEW

**Comment by rjmccall** [comment]
@swift-ci Please test and merge.

**Comment by gottesmm** [comment]
Hmmm... is it possible to write a unit test for this?

**Comment by gottesmm** [comment]
Sorry read the commit stat wrong! Sorry for the noise!

**Comment by rjmccall** [comment]
Actually, I can do much better than that as a unit test.

**Comment by rjmccall** [comment]
@swift-ci Please test

**Review by atrick — APPROVED** [review]
👍

File: unittests/runtime/Exclusivity.cpp [hunk 1]
```diff
@@ -54,3 +54,33 @@ TEST(TextExclusivity, testBogusPC) {
...
+  auto begin = [&](unsigned i) {
+    assert(i < N);
+    swift_beginAccess(&vars[i], &scratches[i], ExclusivityFlags::Modify, 0);
+  };
+  auto end = [&](unsigned i) {
+    assert(i < N);
+    swift_endAccess(&scratches[i]);  
+  };
+  auto accessAll = [&] {
+    for (unsigned i = 0; i != n; ++i) begin(i);
```
Comment (atrick) [inline unittests/runtime/Exclusivity.cpp]: `i != N`

File: unittests/runtime/Exclusivity.cpp [hunk 2]
```diff
@@ -54,3 +54,33 @@ TEST(TextExclusivity, testBogusPC) {
...
+    for (unsigned i = 0; i != n; ++i) end(i);
```
Comment (atrick) [inline unittests/runtime/Exclusivity.cpp]: `i != N`

**Review by atrick — CHANGES_REQUESTED** [review]
Just these two typos...

**Comment by rjmccall** [comment]
Hazards of lazily shooting revisions off for the bots to handle.

**Comment by rjmccall** [comment]
@swift-ci Please test.

**Comment by rjmccall** [comment]
@swift-ci Please test.

**Review by atrick — APPROVED** [review]

**Comment by rjmccall** [comment]
@swift-ci Please test OS X.

**Review by atrick — COMMENTED** [review]
👍 and🥇 for CCC.

**Pull request merged**